### ISSUE TITLE
[sklearn#34669] Document tie-breaking behavior for max_categories in OneHotEncoder/OrdinalEncoder infrequent category grouping

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#295]
**Mirrored from [scikit-learn#34669](https://github.com/scikit-learn/scikit-learn/issues/34669)**

Original title: Document tie-breaking behavior for max_categories in OneHotEncoder/OrdinalEncoder infrequent category grouping
Original labels: Documentation, Needs Triage
Created: 2026-08-06T14:51:19Z

---

> [!WARNING]
> This issue is not yet ready for a PR. If you are interested in contributing to scikit-learn, please have a look at our [contributing guidelines](https://scikit-learn.org/dev/developers/contributing.html), and in particular the sections for [new contributors](https://scikit-learn.org/dev/developers/contributing.html#new-contributors) and the ["Needs triage"](https://scikit-learn.org/dev/developers/contributing.html#issues-tagged-needs-triage) label.

### Describe the issue linked to the documentation

When `max_categories` forces additional categories into the infrequent group beyond what `min_frequency` alone would select, and multiple categories are tied in count at the cutoff boundary, the docs for `max_categories` don't specify which tied category is kept as frequent vs. grouped as infrequent.

Looking at `_identify_infrequent()` in `sklearn/preprocessing/_encoders.py`, the tie-break is implicit rather than documented:

```python
# stable sort to preserve original count order
smallest_levels = np.argsort(category_count, kind="mergesort")[:-frequent_category_count]
infrequent_mask[smallest_levels] = True
```

Counts are sorted using `np.argsort(category_count, kind="mergesort")` — a stable sort — so ties are resolved by the order categories appear in `categories_[i]`, which for `categories='auto'` is lexicographic/numeric sort order. This behavior is not mentioned in the `max_categories` docstring or the User Guide section on infrequent categories (`encoder_infrequent_categories`).

### Suggest a potential alternative/fix

Add a sentence to the `max_categories` parameter docs (in both `OneHotEncoder` and `OrdinalEncoder`) and to the User Guide section on infrequent categories, stating that ties in category count are broken using the sort order of the categories (via a stable sort). This is also relevant to the discussion in #32161.

### Minimal reproducer

```python
import numpy as np
from sklearn.preprocessing import OneHotEncoder

X = np.array([["a"] * 2 + ["b"] * 2 + ["c"] * 5], dtype=object).T
enc = OneHotEncoder(max_categories=2, sparse_output=False).fit(X)
print(enc.infrequent_categories_)
# 'a' and 'b' tie at count=2 -- which one is grouped as infrequent
# is not documented, though it is deterministic.
```

### Additional context

Related: #32161 discusses tie-breaking by lexicographical order for encoder category ordering more broadly.

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Addressed in upstream PR(s). See scikit-learn/scikit-learn PRs for #34669.
